lms/djangoapps/bookmarks_api/views.py

```diff
@@ -0,0 +1,210 @@
+"""
+For more information, see:
+https://openedx.atlassian.net/wiki/display/TNL/Bookmarks+API
+"""
+import logging
+
+from django.core.exceptions import ObjectDoesNotExist, ValidationError, MultipleObjectsReturned
+from django.utils.translation import ugettext as _
+from django.http import Http404
+
+
+from rest_framework import status
+from rest_framework import permissions
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework.generics import ListCreateAPIView
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+from bookmarks_api import serializers
+from openedx.core.lib.api.serializers import PaginationSerializer
+
+from xmodule.modulestore.exceptions import ItemNotFoundError
+from xmodule.modulestore.django import modulestore
+
+from .models import Bookmark
+from .api import get_bookmark
+
+
+log = logging.getLogger(__name__)
+
+
+class BookmarksView(ListCreateAPIView):
+    """
+    List all bookmarks or create.
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    paginate_by = 1000
```

Comment:
The caller should explicitly specify what number of bookmarks it wants in the first page using `page_size` GET parameter.